pyccel/ast/numpy_wrapper.py

```diff
@@ -0,0 +1,289 @@
+                        ('complex',8)  : numpy_cdouble_type,
+                        ('complex',16) : numpy_clongdouble_type}
+
+# Needed to check for numpy arguments type
+Numpy_Bool_ref       = Variable(dtype=NativeVoid(),  name = 'Bool')
+Numpy_Int8_ref       = Variable(dtype=NativeVoid(),  name = 'Int8')
+Numpy_Int16_ref      = Variable(dtype=NativeVoid(),  name = 'Int16')
+Numpy_Int32_ref      = Variable(dtype=NativeVoid(),  name = 'Int32')
+Numpy_Int64_ref      = Variable(dtype=NativeVoid(),  name = 'Int64')
+Numpy_Float_ref      = Variable(dtype=NativeVoid(),  name = 'Float32')
+Numpy_Double_ref     = Variable(dtype=NativeVoid(),  name = 'Float64')
+Numpy_Complex64_ref  = Variable(dtype=NativeVoid(),  name = 'Complex64')
+Numpy_Complex128_ref = Variable(dtype=NativeVoid(),  name = 'Complex128')
+
+numpy_type_check_registry = {
+    (NativeInteger(), 4)       : Numpy_Int32_ref,
+    (NativeInteger(), 8)       : Numpy_Int64_ref,
+    (NativeInteger(), 2)       : Numpy_Int16_ref,
+    (NativeInteger(), 1)       : Numpy_Int8_ref,
+    (NativeReal(), 8)          : Numpy_Double_ref,
+    (NativeReal(), 4)          : Numpy_Float_ref,
+    (NativeComplex(), 4)       : Numpy_Complex64_ref,
+    (NativeComplex(), 8)       : Numpy_Complex128_ref,
+    (NativeBool(), 4)          : Numpy_Bool_ref
+}
+
+# helpers
+def find_in_numpy_dtype_registry(var):
+    """ Find the numpy dtype key for a given variable
+    """
+    dtype = var.dtype
+    prec  = var.precision
+    try :
+        return numpy_dtype_registry[(dtype, prec)]
+    except KeyError:
+        errors.report(PYCCEL_RESTRICTION_TODO,
+                symbol = "{}[kind = {}]".format(dtype, prec),
+                severity='fatal')
+
+# check array
+def Check_Array(py_variable, c_variable, language = 'Fortran'):
+    """
+    Create FunctionCall responsible of checking numpy array order and rank
+    Parameters:
+    ----------
+    c_variable  : Variable
+        The variable needed for the generation of the type check
+    py_variable : Variable
+        The python argument of the check function
+    language    : string
+        Needed to collect the flag for order check
+    Returns
+    -------
+    FunctionCall : Check type FunctionCall
+    """
+    rank = c_variable.rank
+
+    if rank > 1 and language == 'fortran':
+        if c_variable.order == 'F':
+            flag = numpy_flag_f_contig
+        else:
+            flag = numpy_flag_c_contig
+    else:
+        flag = LiteralInteger(0)
+
+    func = FunctionDef(name = 'Check_Array',
+                       arguments = [Variable(dtype = PyccelPyObject(), name = 'a', is_pointer = True),
+                                    Variable(dtype = NativeInteger(),  name = 'rank'),
+                                    Variable(dtype = NativeInteger(),  name = 'flag')],
+                       body      = [],
+                       results   = [Variable(dtype = PyccelPyArrayObject(), name = 'array', is_pointer = True)])
```

Comment:
Does the `FunctionDef` need to be recreated every time we call `Check_array`? I don't see anything that depends on the arguments

Reply:
yeah it can be global one and used in functionCall with the corresponding arguments